packages/preview-service/tests/hooks.ts

```diff
@@ -0,0 +1,70 @@
+import { startServer } from '../server/server'
+import { getTestDb } from './testKnex'
+import { testLogger as logger } from '../observability/logging'
+import type { Knex } from 'knex'
+
+const testDb = getTestDb()
+
+const truncateTables = async (params: { db: Knex; tableNames?: string[] }) => {
+  const { db } = params
+  let { tableNames } = params
+
+  if (!tableNames?.length) {
+    //why is server config only created once!????
+    // because its done in a migration, to not override existing configs
+    const protectedTables = ['server_config']
+    // const protectedTables = [ 'server_config', 'user_roles', 'scopes', 'server_acl' ]
+    tableNames = (
+      await db('pg_tables')
+        .select('tablename')
+        .where({ schemaname: 'public' })
+        .whereRaw("tablename not like '%knex%'")
+        .whereNotIn('tablename', protectedTables)
+    ).map((table: { tablename: string }) => table.tablename)
+  }
+
+  if (!tableNames || !tableNames.length) return
+
+  await db.raw(`truncate table ${tableNames.join(',')} cascade`)
```

Comment:
Doesn't all of this happen in the same DB that `server` works off of, at least locally? We def wouldn't want to truncate/delete all of that data

Reply:
Even if this is only meant for a test DB, I just want to be sure that there's no way this can accidentally run on the main local speckle server DB and wipe everything there